cli/build.rs

```diff
@@ -99,14 +101,35 @@ fn main() {
     deno_typescript::op_fetch_asset(custom_libs),
   );
 
-  deno_typescript::mksnapshot_bundle_ts(
-    &mut runtime_isolate,
-    &snapshot_path,
-    &bundle_path,
-    &main_module_name,
-  )
-  .expect("Failed to create snapshot");
+  js_check(
+    runtime_isolate.execute("typescript.js", deno_typescript::TYPESCRIPT_CODE),
+  );
+
+  let mut sorted_files = std::fs::read_dir("js2/")
+    .unwrap()
+    .map(|dir_entry| {
+      let file = dir_entry.unwrap();
+      let filename = file.path().to_string_lossy().to_string();
+      filename
+    })
+    .filter(|filename| filename.ends_with(".js"))
+    .collect::<Vec<String>>();
+
+  sorted_files.sort();
+
+  eprintln!("sorted files {:#?}", sorted_files);
+  for file in sorted_files {
+    println!("cargo:rerun-if-changed={}", file);
+    js_check(
+      runtime_isolate.execute(&file, &std::fs::read_to_string(&file).unwrap()),
+    );
+  }
 
+  let snapshot = runtime_isolate.snapshot();
+  let snapshot_slice: &[u8] = &*snapshot;
+  println!("Snapshot size: {}", snapshot_slice.len());
+  std::fs::write(&compiler_snapshot_path, snapshot_slice).unwrap();
+  println!("Snapshot written to: {} ", compiler_snapshot_path.display());
```

Comment:
Abstract snapshot creation into a new function so these println lines are not repeated.

Probably something like?
```
fn create_snapshot(files: Vec<Path>)
```